packages/storybook/src/stories/4-ContractEditor.stories.js

```diff
@@ -71,27 +112,77 @@ This is text. This is *italic* text. This is **bold** text. This is a [link](htt
             },
             ...extraText.document.children
           ]
+          store.dispatch(addTemplate(template))
           Transforms.insertNodes(editor, slateClause, { at: Editor.end(editor, [])});
         });
     }
   }, [templateUrl, markdownText, editor]);
 
   const onContractChange = useCallback((value) => {
     setSlateValue(value);
-    action('contract-changed')(value);
-  }, []);
+    action('Contract -> Change: ')(value);
+  }, [editor]);
 
   const clausePropsObject = {
-    CLAUSE_DELETE_FUNCTION: action('clause-deleted'),
-    CLAUSE_EDIT_FUNCTION: action('clause-edit'),
-    CLAUSE_TEST_FUNCTION: action('clause-test')
+    CLAUSE_DELETE_FUNCTION: action('Clause -> Deleted'),
+    CLAUSE_EDIT_FUNCTION: action('Clause -> Edit'),
+    CLAUSE_TEST_FUNCTION: action('Clause -> Test')
   };
 
   const augmentEditor = useCallback((slateEditor) => {
     setEditor(slateEditor);
+    store.dispatch(addEditor(slateEditor))
     return slateEditor;
   }, []);
 
+  const parseClause = useCallback(async (val) => {
+    const newReduxState = store.getState();
+    const value = {
+      document: {
+        children: val.children
+      }
+    };
+    const text = slateTransformer.toMarkdown(value);
+
+    const SLICE_INDEX_1 = val.data.src.lastIndexOf('/') + 1;
+    const SLICE_INDEX_2 = val.data.src.indexOf('@');
+    const TEMPLATE_NAME = val.data.src.slice(SLICE_INDEX_1, SLICE_INDEX_2);
+    const ciceroClause = new Clause(newReduxState[TEMPLATE_NAME]);
+
+    ciceroClause.parse(text);
+
+    const ast = ciceroClause.getData();
+    const something = await ciceroClause.draft({format:'slate'});
+
+    const found = val.children[1].children.filter(element => element.type === 'formula' && element.data.name === 'formula');
+    
+    action('Clause -> Parse: ')({
+      'Clause': ciceroClause,
+      'AST': ast,
+      'Draft': something
+    });
+
+    /* XXX What do we do with this? - JS
```

Comment:
Unsure about this. It's not commented out in [my branch](https://github.com/accordproject/web-components/blob/irmerk/i69/formulae-ui/packages/storybook/src/stories/4-ContractEditor.stories.js#L174)

Reply:
I know! But it seems to assume a specific template, no? Do we really want to leave this code in for the release?

Reply:
I don't think so - just will need to come back to it.